7-extract_plier_pathways.R

```diff
@@ -67,41 +66,80 @@ all.paths <- PLIER::combinePaths(bloodCellMarkersIRISDMAP,
 #### Function for converting column to row names -------------------------------
 
 convert_row_names <- function(expr, cancer_type){
+  # If the cancer type is GBM, convert ENSG to gene symbols
+  # then convert the gene column to rownames
+  #
+  # Inputs: gene expression matrix with genes in first column, and cancer type
+  # Returns: modified gene expression matrix with gene names as row names
+  
   if (cancer_type == "GBM") {
     expr <- expr %>%
       mutate(gene = ensembldb::select(EnsDb.Hsapiens.v86::EnsDb.Hsapiens.v86,
                                       keys = as.character(gene),
                                       keytype = "GENEID",
                                       columns = "SYMBOL")$SYMBOL)
   }
+  
   column_to_rownames(expr,
                      var = "gene")
+  
+}
+
+#### Function to get jaccard values from a list of PLIER results ---------------
+
+return_plier_jaccard <- function(test_PLIER, array_silver, seq_silver){
+  # Given a set of PLIER results (which is a list), compare significant pathways
+  # to two silver sets of pathways defined by array and RNA-seq data only
+  # Jaccard similaritiy is defined as O(intersect)/O(union).
+  #
+  # Inputs: PLIER result, pathway set 1, pathway set 2
+  # Returns: data frame with two rows (array, seq) with stats for each overlap
+  
+  test_genes <- test_PLIER[["summary"]] %>%
+    filter(FDR < 0.05) %>%
+    pull(pathway) %>%
+    unique()
+  
+  array_jaccard <- length(intersect(array_silver, test_genes))/length(union(array_silver, test_genes))
+  seq_jaccard <- length(intersect(seq_silver, test_genes))/length(union(seq_silver, test_genes))  
+  
+  data.frame(silver = c("array", "seq"),
+             n_silver = c(length(array_silver),
+                          length(seq_silver)),
+             n_test = length(test_genes),
+             n_intersect = c(length(intersect(array_silver, test_genes)),
+                             length(intersect(seq_silver, test_genes))),
+             n_union = c(length(union(array_silver, test_genes)),
+                         length(union(seq_silver, test_genes))),
+             n_common_genes = nrow(test_PLIER[["Z"]]),
+             k = ncol(test_PLIER[["Z"]]),
```

Comment:
Why Z here?

Reply:
I may have misinterpreted Figure 1 here. I'm trying to get the at the number of genes and LVs used in that particular PLIER result. https://www.nature.com/articles/s41592-019-0456-1

Reply:
I'm glad I asked because my brain said "this is looking at the z-score as normalized data only"